src/io/obf.rs

```diff
@@ -220,17 +221,39 @@ pub async fn obfuscate<M, SU, SH, ST, R, P>(
             {
                 player.play_music(format!("bgm/obf_bgm{}.mp3", (2 * level + num) % 3 + 2));
             }
-            let r = &public_data.r[num];
-            log_mem(format!("Computed R ({},{}) (n+1)x(n+1)", r.row_size(), r.col_size()));
+
+            let tag = format!("S_{}_{}", level, num).into_bytes();
+            let s_i_bar =
+                hash_sampler.sample_hash(&params, hash_key, &tag, d, d, DistType::BitDist);
+            let s_i_num = s_i_bar.concat_diag(&[&one_identity]);
+
+            #[cfg(feature = "debug")]
+            handles.push(store_and_drop_matrix(
+                s_i_num.clone(),
+                &dir_path,
+                &format!("s_{}_{}", level, num),
+            ));
+
+            log_mem(format!(
+                "Computed S ({},{}) (n+1)x(n+1)",
```

Comment:
I think that the size n+1 x n+1 refers to the LWE setting, here it should be d+1 x d+1